### PR TITLE
Fix vpn, oauth for AWS

### PR DESCRIPTION
**Pull request by csrwng**
Fixes needed for 4.3 to work on AWS (vpn, oauth)

### CODE REVIEW

File: assets/ca-operator/ca-operator-configmap.yaml [hunk 1]
```diff
@@ -3,5 +3,5 @@ kind: ConfigMap
 metadata:
   name: ca-operator
 data:
-  initial-ca.crt: |-
+  initial-ca.crt: |
```
Comment (csrwng) [inline assets/ca-operator/ca-operator-configmap.yaml:6]: With the trailing `-` the final line feed was getting truncated. This resulted in an incorrectly formatted ca when `cat`ed by the ca-opertor pod.

File: assets/kube-apiserver/kube-apiserver-configmap.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ data:
   aggregator-client-ca.crt: |-
 {{ include_pki "root-ca.crt" 4 }}
   kubelet-client-ca.crt: |-
-{{ include_pki "root-ca.crt" 4 }}
+{{ include_pki "combined-ca.crt" 4 }}
```
Comment (csrwng) [inline assets/kube-apiserver/kube-apiserver-configmap.yaml:9]: the kubelet is serving with a cert signed by the `cluster-signer` CA, the `combined-ca` is needed for the client to properly validate the kubelet cert.

File: assets/oauth-openshift/oauth-server-config.yaml [hunk 3]
```diff
@@ -26,9 +26,7 @@ oauthConfig:
     method: deny
     serviceAccountMethod: prompt
 {{ if .IdentityProviders }}  identityProviders:
-{{ .IdentityProviders | indent 2 }}
-{{- else }}  identityProviders: []
-{{- end -}}
+{{ trimTrailingSpace .IdentityProviders | indent 2 }}{{- else }}  identityProviders: []{{- end }}
```
Comment (csrwng) [inline assets/oauth-openshift/oauth-server-config.yaml:29]: In the case where `.IdentityProviders` was empty, the trailing `-` in the final `{{ end }}` was causing the next line (starting with `loginURL` to get appended to identityProviders, resulting in invalid YAML. With this change, the `.IdentityProviders` is trimmed of a final newline, and the newline is provided by the template. Tested with/without a value for .IdentityProviders and worked as expected with both.

**Comment by csrwng** [comment]
@relyt0925 @KodieGlosserIBM ptal
Added comments where there are changes that could impact your use case.

**Comment by relyt0925** [comment]
testing

**Review by relyt0925 — APPROVED** [review]

**Pull request merged**